src/App.tsx

```diff
@@ -1,11 +1,87 @@
 import './installSesLockdown';
-import 'ses';
+import { motion } from 'framer-motion';
+import { useEffect } from 'react';
+import { useAtom } from 'jotai';
+import { E, ERef } from '@endo/eventual-send';
+import { makeAsyncIterableFromNotifier as iterateNotifier } from '@agoric/notifier';
+
+import {
+  brandToInfoAtom,
+  walletAtom,
+  pursesAtom,
+  offersAtom,
+} from 'store/store';
+import WalletConnection from 'components/WalletConnection';
+import { INTER_LOGO } from 'assets/assets';
+import Swap from 'components/Swap';
+
+const watchPurses = async (
+  wallet: ERef<any>,
+  setPurses: (purses: any) => void,
+  mergeBrandToInfo: (entries: Iterable<Iterable<any>>) => void
+) => {
+  const n = await E(wallet).getPursesNotifier();
+  for await (const purses of iterateNotifier(n)) {
+    setPurses(purses);
+    purses.forEach(
+      ({
+        brand,
+        displayInfo,
+        brandPetname: petname,
+      }: {
+        brand: any;
+        displayInfo: any;
+        brandPetname: string;
+      }) => {
+        const decimalPlaces = displayInfo && displayInfo.decimalPlaces;
+        const assetKind = displayInfo && displayInfo.assetKind;
+
+        const newInfo = {
+          petname,
+          assetKind,
+          decimalPlaces,
+        };
+
+        mergeBrandToInfo([[brand, newInfo]]);
+      }
+    );
+  }
+};
+
+const watchOffers = async (wallet: any, setOffers: (offers: any) => void) => {
+  const offerNotifier = E(wallet).getOffersNotifier();
+  for await (const offers of iterateNotifier(offerNotifier)) {
+    setOffers(offers);
+  }
+};
 
 const App = () => {
+  const [wallet] = useAtom(walletAtom);
+  const [_brandToInfo, mergeBrandToInfo] = useAtom(brandToInfoAtom);
+  const [_purses, setPurses] = useAtom(pursesAtom);
+  const [_offers, setOffers] = useAtom(offersAtom);
+
+  useEffect(() => {
+    if (wallet === null) return;
+
+    watchPurses(wallet, setPurses, mergeBrandToInfo).catch((err: Error) =>
+      console.error('got watchPurses err', err)
```

Comment:
will need to improve error handling at some point

Reply:
Do you think a toast or something suffices?

Reply:
toast would suffice! just something so the user knows there's a problem. we can improve the feedback or recovery over time

Reply:
SGTM. Added a TODO